## 🌍 Introduction

Climate change is one of the most important global challenges of our time. 🌱 Governments, researchers, and organizations around the world are increasingly using data to better understand how human activities affect the environment. By analyzing greenhouse gas emissions across countries, industries, and time periods, we can gain valuable insights into the patterns and trends that shape our planet’s climate. 📊

In this project, we explore a dataset containing quarterly indicators related to greenhouse gas emissions. The data allows us to observe how emissions evolve over time, how different industries contribute to environmental impact, and how patterns vary across countries and economic regions. Looking at these trends can help highlight which sectors play the largest role in emissions and where efforts toward sustainability might have the greatest impact. 🌎

To guide this analysis, the following key questions are considered:

🔎 **Research Questions**

* How have greenhouse gas emissions changed over time across countries and industries?
* Which industries contribute the most to overall emissions levels?
* Are there noticeable differences in emissions patterns between countries or economic regions?
* How do different gas types contribute to total emissions trends?

By answering these questions, this analysis aims to uncover meaningful insights from the data and better understand how economic activity and industrial sectors influence global emissions patterns. 📈✨

---

## 📂 Data Source

The dataset used in this project was obtained from the **Climate Change Indicators Dashboard** maintained by the International Monetary Fund.

This platform compiles climate-related indicators from multiple international sources to support research, policy development, and economic analysis related to climate change. 🌐

**Source:**
[https://climatedata.imf.org/datasets/543872e1d86c49e3a3bdf38f2b758f92/about](https://climatedata.imf.org/datasets/543872e1d86c49e3a3bdf38f2b758f92/about)

### 📥 Loading the Dataset and Libraries

Before we start, we need to install the necessary Python libraries and **load the dataset**.


In [25]:
# Install the required libraries
%pip install pandas numpy matplotlib seaborn scikit-learn -q

# Install and update the watermark package to display environment and library version information
%pip install -q -U watermark

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 25.2 -> 26.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 25.2 -> 26.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [26]:
# Data manipulation and visualization
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

# Pre-Processing and Machine Learning
from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.preprocessing import StandardScaler
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import LogisticRegression
from sklearn.pipeline import Pipeline
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix
import joblib

In [27]:
# Load the watermark extension
%reload_ext watermark

# Display metadata for your notebook
%watermark -a "Maykon - 💬 Sentiment Analysis Project" -d -u -v -p numpy,pandas,matplotlib,seaborn

Author: Maykon - 💬 Sentiment Analysis Project

Last updated: 2026-03-20

Python implementation: CPython
Python version       : 3.13.7
IPython version      : 9.11.0

numpy     : 2.4.3
pandas    : 3.0.1
matplotlib: 3.10.8
seaborn   : 0.13.2



In [28]:
# Load the Sentiment Analysis dataset from a local CSV file into a pandas DataFrame
data = r"data\dataset.csv"

df = pd.read_csv(data)

In [29]:
# It returns the dimensions of the DataFrame as a tuple
df.shape

(2388, 75)

In [30]:
df.head(10) #Displays the first rows of the DataFrame df.

,ObjectId,Country,ISO2,ISO3,Indicator,Unit,Source,CTS Code,CTS Name,CTS Full Descriptor,...,2022Q4,2023Q1,2023Q2,2023Q3,2023Q4,2024Q1,2024Q2,2024Q3,2024Q4,2025Q1
0,1,Advanced Economies,NaN,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,64.688709,61.319185,65.867074,68.262567,66.039285,62.452184,73.597480,75.162258,71.925035,70.263853
1,2,Advanced Economies,NaN,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,64.104215,64.183297,65.752729,65.954354,65.597731,65.374440,72.340776,72.275717,72.779925,74.291176
2,3,Advanced Economies,NaN,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,0.248269,0.237558,0.236841,0.235717,0.241366,0.233355,0.236395,0.238220,0.242984,0.235360
3,4,Advanced Economies,NaN,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,0.242981,0.240553,0.238109,0.236431,0.236390,0.235936,0.237543,0.239370,0.238111,0.239349
4,5,Advanced Economies,NaN,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,386.112317,426.972747,320.899060,297.709005,381.070358,428.160874,324.106184,300.675095,380.802186,453.832593
5,6,Advanced Economies,NaN,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,356.181839,351.659954,360.585785,356.274024,358.131407,362.286817,361.624871,366.993041,362.452924,366.114987
6,7,Advanced Economies,NaN,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,161.104979,162.432331,161.521244,162.742242,161.812556,161.213234,159.000930,159.576325,158.233082,158.150828
7,8,Advanced Economies,NaN,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,161.160560,161.888495,162.344945,162.440571,161.834364,160.805631,159.787676,159.143909,158.294445,157.729357
8,9,Advanced Economies,NaN,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,160.070361,202.983673,93.273901,66.468479,152.977150,204.262101,91.271379,65.698292,150.401085,225.182552
9,10,Advanced Economies,NaN,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,130.674082,125.347610,132.250003,127.642668,130.462923,135.870810,129.258876,135.

In [31]:
df.sample(10) # Random 15 rows

,ObjectId,Country,ISO2,ISO3,Indicator,Unit,Source,CTS Code,CTS Name,CTS Full Descriptor,...,2022Q4,2023Q1,2023Q2,2023Q3,2023Q4,2024Q1,2024Q2,2024Q3,2024Q4,2025Q1
1726,1741,Other Oceania sub-regions,NaN,NAOOSR,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,0.034632,0.034778,0.034914,0.035042,0.035161,0.035250,0.035333,0.035411,0.035485,0.035569
1500,1515,Northern Europe,NaN,NANE,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,0.276987,0.268823,0.266194,0.269903,0.279995,0.274113,0.275711,0.279433,0.286795,0.281512
915,930,Europe,NaN,EURTMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,4.222404,4.198414,4.158808,4.094559,4.075782,4.132840,4.165890,4.171573,4.213785,4.258478
503,518,Central Asia,NaN,NACA,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,0.748122,0.747168,0.753613,0.744107,0.750167,0.749915,0.750922,0.786455,0.752721,0.752031
2280,2409,Western Europe,NaN,NAWE,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,1.459429,1.418046,1.414118,1.400379,1.449512,1.412271,1.415089,1.408418,1.452072,1.413501
2200,2215,Western Europe,NaN,NAWE,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,0.205419,0.197698,0.197155,0.194169,0.202913,0.196849,0.198241,0.196108,0.203684,0.197612
207,222,Americas,NaN,AMETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,10.209623,10.499669,10.292605,10.040220,9.757369,9.701180,9.775153,9.610651,9.331494,9.557888
1483,1498,Northern Europe,NaN,NANE,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,7.593313,7.489029,7.251631,7.210143,7.070108,7.041330,6.963210,6.943717,7.093549,7.053362
164,179,Africa,NaN,NA605,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,803.004912,808.973790,797.203341,804.753136,827.274026,817.492766,802.584809,822.763378,836.712761,841.829336
2296,2425,World,NaN,WLD,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,525.899637,570.234774,458.589698,434.635386,525.493569,580.055848,467.721540,445.342093,534.770384,619.373353


In [32]:
df.tail() #Displays the last 5 rows of the DataFrame df.

,ObjectId,Country,ISO2,ISO3,Indicator,Unit,Source,CTS Code,CTS Name,CTS Full Descriptor,...,2022Q4,2023Q1,2023Q2,2023Q3,2023Q4,2024Q1,2024Q2,2024Q3,2024Q4,2025Q1
2383,2512,World,NaN,WLD,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,461.597897,463.399306,467.858827,472.314308,475.491386,477.755273,482.105839,489.264542,490.408500,494.987684
2384,2513,World,NaN,WLD,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,385.834429,393.086534,387.666238,387.794684,399.030287,406.843232,401.637057,398.964513,412.201171,424.677965
2385,2514,World,NaN,WLD,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,384.308985,386.096720,389.671794,394.225184,397.584046,398.709245,403.300465,405.857906,410.869060,416.101641
2386,2515,World,NaN,WLD,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,29.280399,29.322187,29.068810,29.581099,29.845763,30.088260,29.822110,30.175253,30.450210,30.837705
2387,2516,World,NaN,WLD,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...",...,29.114959,29.104364,29.322747,29.662102,29.728647,29.899336,30.012889,30.204904,30.369061,30.660960


#### 🔎📊 Exploratory Data Analysis (EDA)

This is where we start exploring the dataset to understand how greenhouse gas emissions behave across countries, industries, and time 🌍📊. The goal here is to identify patterns, trends, and potential anomalies that can guide the rest of the analysis 💡

During this phase, we use visualizations 📈 such as time series plots, comparisons between industries 🏭, and country-level analyses 🌎 to better understand how emissions evolve over time. We also rely on summary statistics to examine the distribution of values, detect missing data, and identify any inconsistencies in the dataset 🛠️

By performing this initial exploration, we build a solid understanding of the data, which helps ensure that any further analysis, transformations, or modeling steps are accurate and meaningful 🚀

In [33]:
# Information about the dataframe
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2388 entries, 0 to 2387
Data columns (total 75 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   ObjectId             2388 non-null   int64  
 1   Country              2388 non-null   str    
 2   ISO2                 0 non-null      float64
 3   ISO3                 2388 non-null   str    
 4   Indicator            2388 non-null   str    
 5   Unit                 2388 non-null   str    
 6   Source               2388 non-null   str    
 7   CTS Code             2388 non-null   str    
 8   CTS Name             2388 non-null   str    
 9   CTS Full Descriptor  2388 non-null   str    
 10  Industry             2388 non-null   str    
 11  Gas Type             2388 non-null   str    
 12  Seasonal Adjustment  2388 non-null   str    
 13  Scale                2388 non-null   str    
 14  2010Q1               2388 non-null   float64
 15  2010Q2               2388 non-null   float64
 16 

In [34]:
df.describe(include='all').T  # Generates descriptive statistics for all numeric columns in your DataFrame.

,count,unique,top,freq,mean,std,min,25%,50%,75%,max
ObjectId,2388.0,NaN,NaN,NaN,1213.093802,700.517297,1.0,611.75,1208.5,1805.25,2516.0
Country,2388,25,Advanced Economies,100,NaN,NaN,NaN,NaN,NaN,NaN,NaN
ISO2,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
ISO3,2388,25,AETMP,100,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Indicator,2388,1,Quarterly greenhouse gas (GHG) air emissions a...,2388,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...
2024Q1,2388.0,NaN,NaN,NaN,222.718239,890.758714,-0.0,0.83503,12.572202,87.390084,13405.758
2024Q2,2388.0,NaN,NaN,NaN,214.658705,855.148067,-0.338902,0.829891,11.940605,83.978671,13120.4639
2024Q3,2388.0,NaN,NaN,NaN,224.904458,905.813811,-0.0,0.851089,12.314523,84.015516,13540.92567
2024Q4,2388.0,NaN,NaN,NaN,223.765768,894.538794,-0.0,0.846428,12.659841,88.658769,13516.17495


##### Check for missing values on the DataFrame:

In [35]:
df.isna().sum()  # You can also use df.isnull().sum() — both do the same thing

ObjectId        0
Country         0
ISO2         2388
ISO3            0
Indicator       0
             ... 
2024Q1          0
2024Q2          0
2024Q3          0
2024Q4          0
2025Q1          0
Length: 75, dtype: int64

In [36]:
# Drop the all‑null ISO2 column. SO2 has 100% missing values → i have to drop this column, it won’t be useful.
df = df.drop(columns=['ISO2', 'ObjectId'])

In [37]:
df.columns

Index(['Country', 'ISO3', 'Indicator', 'Unit', 'Source', 'CTS Code',
       'CTS Name', 'CTS Full Descriptor', 'Industry', 'Gas Type',
       'Seasonal Adjustment', 'Scale', '2010Q1', '2010Q2', '2010Q3', '2010Q4',
       '2011Q1', '2011Q2', '2011Q3', '2011Q4', '2012Q1', '2012Q2', '2012Q3',
       '2012Q4', '2013Q1', '2013Q2', '2013Q3', '2013Q4', '2014Q1', '2014Q2',
       '2014Q3', '2014Q4', '2015Q1', '2015Q2', '2015Q3', '2015Q4', '2016Q1',
       '2016Q2', '2016Q3', '2016Q4', '2017Q1', '2017Q2', '2017Q3', '2017Q4',
       '2018Q1', '2018Q2', '2018Q3', '2018Q4', '2019Q1', '2019Q2', '2019Q3',
       '2019Q4', '2020Q1', '2020Q2', '2020Q3', '2020Q4', '2021Q1', '2021Q2',
       '2021Q3', '2021Q4', '2022Q1', '2022Q2', '2022Q3', '2022Q4', '2023Q1',
       '2023Q2', '2023Q3', '2023Q4', '2024Q1', '2024Q2', '2024Q3', '2024Q4',
       '2025Q1'],
      dtype='str')

In [38]:
df.dtypes

Country          str
ISO3             str
Indicator        str
Unit             str
Source           str
              ...   
2024Q1       float64
2024Q2       float64
2024Q3       float64
2024Q4       float64
2025Q1       float64
Length: 73, dtype: object

In [39]:
# Identify all columns that are NOT quarter columns (i.e., identifiers)
id_vars = [col for col in df.columns if not col.startswith(('2010','2011','2012','2013','2014','2015','2016','2017','2018','2019','2020','2021','2022','2023','2024','2025'))]

# Quarter columns are the rest
quarter_cols = [col for col in df.columns if col not in id_vars]

# Melt the dataframe
df_long = pd.melt(df,
                  id_vars=id_vars,
                  value_vars=quarter_cols,
                  var_name='Quarter',
                  value_name='Emissions')

# Check the result
print("New shape:", df_long.shape)
print(df_long.head())

New shape: (145668, 14)
              Country   ISO3  \
0  Advanced Economies  AETMP   
1  Advanced Economies  AETMP   
2  Advanced Economies  AETMP   
3  Advanced Economies  AETMP   
4  Advanced Economies  AETMP   

                                           Indicator  \
0  Quarterly greenhouse gas (GHG) air emissions a...   
1  Quarterly greenhouse gas (GHG) air emissions a...   
2  Quarterly greenhouse gas (GHG) air emissions a...   
3  Quarterly greenhouse gas (GHG) air emissions a...   
4  Quarterly greenhouse gas (GHG) air emissions a...   

                                    Unit  \
0  Million metric tons of CO2 equivalent   
1  Million metric tons of CO2 equivalent   
2  Million metric tons of CO2 equivalent   
3  Million metric tons of CO2 equivalent   
4  Million metric tons of CO2 equivalent   

                                              Source CTS Code  \
0  Organisation for Economic Co-operation and Dev...    ECNGA   
1  Organisation for Economic Co-operation and Dev..

In [40]:
# Convert Quarter string to datetime (using first day of the quarter)
df_long['Date'] = pd.to_datetime(df_long['Quarter'])

# Check the result
print(df_long[['Quarter', 'Date']].head())

  Quarter       Date
0  2010Q1 2010-01-01
1  2010Q1 2010-01-01
2  2010Q1 2010-01-01
3  2010Q1 2010-01-01
4  2010Q1 2010-01-01


C:\Users\LarTI\AppData\Local\Temp\ipykernel_6344\627629557.py:2: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df_long['Date'] = pd.to_datetime(df_long['Quarter'])


In [41]:
# Check for missing emissions
print("Missing emissions:", df_long['Emissions'].isnull().sum())

# Check unique values in key columns
print("Unique Indicators:", df_long['Indicator'].nunique())
print(df_long['Indicator'].value_counts().head())

print("\nUnique Industries:", df_long['Industry'].nunique())
print(df_long['Industry'].value_counts().head())

print("\nUnique Gas Types:", df_long['Gas Type'].nunique())
print(df_long['Gas Type'].value_counts().head())

print("\nUnique Seasonal Adjustment:", df_long['Seasonal Adjustment'].unique())

Missing emissions: 0
Unique Indicators: 1
Indicator
Quarterly greenhouse gas (GHG) air emissions accounts    145668
Name: count, dtype: int64

Unique Industries: 10
Industry
Manufacturing                    15250
Other Services Industries        15250
Total Industry and Households    15250
Transportation and Storage       14640
Total Households                 14518
Name: count, dtype: int64

Unique Gas Types: 5
Gas Type
Carbon dioxide       30500
Greenhouse gas       30500
Methane              30500
Nitrous oxide        30500
Fluorinated gases    23668
Name: count, dtype: int64

Unique Seasonal Adjustment: <StringArray>
['Not Seasonally Adjusted', 'Seasonally Adjusted']
Length: 2, dtype: str


No missing emissions – that’s excellent and simplifies the analysis.

One indicator – we are working with the quarterly greenhouse gas emissions accounts. Good, no need to filter multiple indicators.

10 industries – the counts per industry are balanced (most appear 15,250 times except Transportation and Storage and Total Households, which are slightly different). This is expected because some industries may not exist in all countries or all gas types.

5 gas types – notice that Fluorinated gases has fewer rows (23,668) than the others (30,500). That likely means not all country–industry–seasonality combinations report fluorinated gases. That’s realistic.

Two seasonal adjustments – you have both “Not Seasonally Adjusted” and “Seasonally Adjusted” versions for each combination. You’ll need to decide which one to use for trend analysis. Often, seasonally adjusted series are preferred to see underlying trends without quarterly fluctuations.

In [42]:
# after clear the data, it's always important to save a copy to avoid lose the data
dfc = df_long.copy()

In [43]:
dfc.head()

,Country,ISO3,Indicator,Unit,Source,CTS Code,CTS Name,CTS Full Descriptor,Industry,Gas Type,Seasonal Adjustment,Scale,Quarter,Emissions,Date
0,Advanced Economies,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...","Agriculture, Forestry and Fishing",Carbon dioxide,Not Seasonally Adjusted,Units,2010Q1,39.798622,2010-01-01
1,Advanced Economies,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...","Agriculture, Forestry and Fishing",Carbon dioxide,Seasonally Adjusted,Units,2010Q1,43.382395,2010-01-01
2,Advanced Economies,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...","Agriculture, Forestry and Fishing",Fluorinated gases,Not Seasonally Adjusted,Units,2010Q1,0.245606,2010-01-01
3,Advanced Economies,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...","Agriculture, Forestry and Fishing",Fluorinated gases,Seasonally Adjusted,Units,2010Q1,0.250689,2010-01-01
4,Advanced Economies,AETMP,Quarterly greenhouse gas (GHG) air emissions a...,Million metric tons of CO2 equivalent,Organisation for Economic Co-operation and Dev...,ECNGA,Greenhouse Gas Emissions (GHG); Air Emissions ...,"Environment, Climate Change, Greenhouse Gas Em...","Agriculture, Forestry and Fishing",Greenhouse gas,Not Seasonally Adjusted,Units,2010Q1,409.375523,2010-01-01


In [44]:
dfc.dtypes

Country                           str
ISO3                              str
Indicator                         str
Unit                              str
Source                            str
CTS Code                          str
CTS Name                          str
CTS Full Descriptor               str
Industry                          str
Gas Type                          str
Seasonal Adjustment               str
Scale                             str
Quarter                           str
Emissions                     float64
Date                   datetime64[us]
dtype: object

We need to keep only the rows where the adjustment type is "Seasonally Adjusted". This removes the "Not Seasonally Adjusted" version.
An then shows the number of rows and columns in the filtered DataFrame. You should see roughly half the original rows (because we originally had two versions). For example, if the original long‑format had ~145,668 rows, after filtering we'll have around 72,834 rows (one per unique combination instead of two).

and after that we need to confirms that now only 'Seasonally Adjusted' remains. If we see only one value, the filter worked.

In [45]:
dfc = dfc[dfc['Seasonal Adjustment'] == 'Seasonally Adjusted']
dfc.shape
dfc['Seasonal Adjustment'].unique()

<StringArray>
['Seasonally Adjusted']
Length: 1, dtype: str

📈 Visualizing Emissions Trends
Now that the data is cleaned, reshaped into long format, and filtered to seasonally adjusted values, we can begin answering our core research questions through visual exploration. Each plot is designed to reveal a different aspect of global greenhouse gas emissions:

How have emissions changed over time? → We’ll start with a global time series to identify long‑term trends and notable events.

Which industries contribute the most? → We’ll compare total emissions across sectors.

How do different gas types behave? → We’ll break down emissions by gas to see which components drive overall trends.

Are there differences between countries or regions? → Later, we’ll map emissions across countries and group them by region.

All visualizations below use the seasonally adjusted version of the data to remove quarterly noise and highlight the underlying patterns. Let’s begin. 🚀